packages/babel-helpers/src/helpers.js

```diff
@@ -1410,6 +1416,36 @@ helpers.classStaticPrivateMethodSet = helper("7.3.2")`
   }
 `;
 
+helpers.classStaticPrivateFieldDestructureSet = helper("7.99.0")`
+  export default function _classStaticPrivateFieldDestructureSet(receiver, classConstructor, descriptor) {
```

Comment:
The more I look at this, I think we can actually reuse this code in the spec instance helpers. Eg, get the descriptor from the instance map, then call over to here. The only change necessary would be for `receiver !== classConstructor` to be changed a bit.

Reply:
Agreed, I will come up with refactors later. Also we should check if `classStaticPrivateFieldDestructureSet` exists because of backward compatibility to older `@babel/helpers`.